x/community/keeper/incentives_test.go

```diff
@@ -134,9 +139,147 @@ func (suite *IncentivesTestSuite) TestStartDisableInflationUpgrade() {
+
+	kavadistAcc := ak.GetModuleAccount(suite.Ctx, kavadisttypes.KavaDistMacc)
+	kavaDistCoinsBefore := suite.App.GetBankKeeper().GetAllBalances(suite.Ctx, kavadistAcc.GetAddress())
+	suite.Equal(
+		fundCoins,
+		kavaDistCoinsBefore,
+		"x/kavadist balance should be funded",
+	)
+
+	// -------------
+	// Run upgrade
+
+	suite.setUpgradeTimeFromNow(-2 * time.Minute)
+	err = suite.Keeper.StartCommunityFundConsolidation(suite.Ctx)
+	suite.NoError(err, "consolidation should not error")
+
+	// -------------
+	// Check results
+
+	suite.Run("balances should be correct", func() {
+		feePoolAfter := suite.App.GetDistrKeeper().GetFeePool(suite.Ctx)
+		suite.Equal(
+			initialFeePoolDust,
+			feePoolAfter.CommunityPool,
+			"x/distribution community pool should be sent to x/community",
+		)
+
+		kavaDistCoinsAfter := suite.App.GetBankKeeper().GetAllBalances(suite.Ctx, kavadistAcc.GetAddress())
+		suite.Equal(
+			sdk.NewCoins(),
+			kavaDistCoinsAfter,
+			"x/kavadist balance should be empty",
+		)
+
+		totalExpectedCommunityPoolCoins := communityBalanceBefore.
+			Add(initialFeePoolCoins...). // x/distribution fee pool
+			Add(fundCoins...)            // x/kavadist module balance
+
+		communityBalanceAfter := suite.App.GetCommunityKeeper().GetModuleAccountBalance(suite.Ctx)
+
+		suite.Equal(
+			totalExpectedCommunityPoolCoins,
+			communityBalanceAfter,
+			"x/community balance should be increased by the truncated x/distribution community pool",
+		)
+	})
+
+	suite.Run("events should be emitted", func() {
+		communityAcc := ak.GetModuleAccount(suite.Ctx, types.ModuleAccountName)
+		distributionAcc := ak.GetModuleAccount(suite.Ctx, distrtypes.ModuleName)
+		kavadistAcc := ak.GetModuleAccount(suite.Ctx, kavadisttypes.KavaDistMacc)
+
+		events := suite.Ctx.EventManager().Events()
+
+		suite.EventsContains(
+			events,
+			sdk.NewEvent(
+				banktypes.EventTypeTransfer,
+				sdk.NewAttribute(banktypes.AttributeKeyRecipient, communityAcc.GetAddress().String()),
+				sdk.NewAttribute(banktypes.AttributeKeySender, distributionAcc.GetAddress().String()),
+				sdk.NewAttribute(sdk.AttributeKeyAmount, initialFeePoolCoins.String()),
+			),
+		)
+
+		suite.EventsContains(
+			events,
+			sdk.NewEvent(
+				banktypes.EventTypeTransfer,
+				sdk.NewAttribute(banktypes.AttributeKeyRecipient, communityAcc.GetAddress().String()),
+				sdk.NewAttribute(banktypes.AttributeKeySender, kavadistAcc.GetAddress().String()),
+				sdk.NewAttribute(sdk.AttributeKeyAmount, kavaDistCoinsBefore.String()),
+			),
+		)
+	})
+}
+
 func (suite *IncentivesTestSuite) setUpgradeTimeFromNow(t time.Duration) {
 	params, found := suite.Keeper.GetParams(suite.Ctx)
 	suite.True(found)
 	params.UpgradeTimeDisableInflation = suite.Ctx.BlockTime().Add(t)
 	suite.Keeper.SetParams(suite.Ctx, params)
 }
+
+// EventsContains asserts that the expected event is in the provided events
+func (suite *IncentivesTestSuite) EventsContains(events sdk.Events, expectedEvent sdk.Event) {
```

Comment:
could refactor or remove the suite dependency and add this to the common test helpers for use across the whole repo?